src/main/java/fi/aalto/cs/apluscourses/model/Submission.java

```diff
@@ -0,0 +1,55 @@
+package fi.aalto.cs.apluscourses.model;
+
+import java.nio.file.Path;
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+
+public class Submission {
+
+  @NotNull
+  private final Exercise exercise;
+  @NotNull
+  private final SubmissionInfo submissionInfo;
+  @NotNull
+  private final Map<String, Path> files;
+  @NotNull
+  private final Group group;
+
+  /**
+   * Constructs a new object instance.
+   *
+   * @param exercise       Exercise.
+   * @param submissionInfo Information for the submission.
+   * @param files          Map from keys to file paths.
+   * @param group          Group in which the submission is made.
+   */
+  public Submission(@NotNull Exercise exercise,
```

Comment:
is it the same `Submission` entity, that supposed to be used to host the results of the submission later on? If it is, then it could probably be extended with the `id` and as seen earlier `hashCode` and `equals`?

Reply:
No, it shouldn't be the same. We discussed that and I think we decided that the other would be called `SubmissionResult`.

Reply:
at this stage `SubmissionResult` represents what is closer logically to `ExerciseSubmissionResult`, not the `Submission` itself.